apps/nova/src/pages/Settings.tsx

```diff
@@ -1,119 +1,37 @@
-  password: {
-    new: string;
-    current: string;
-    confirm: string;
-  };
-};
+import { ProfileForm } from "@/components/Settings/ProfileForm";
+import { redirect } from "react-router-dom";
+import { Route } from "@/types/routes";
+import { Loader, LoadingError } from "@litespace/luna/Loading";
 
 const Settings: React.FC = () => {
   const intl = useFormatMessage();
-  const { user, loading, fetching } = useUser();
-  const toast = useToast();
-  const [photo, setPhoto] = useState<File | null>(null);
-
-  const form = useForm<IForm>({
-    defaultValues: {
-      name: user?.name || "",
-      email: user?.email || "",
-      phoneNumber: user?.phoneNumber || "",
-      password: {
-        new: "",
-        current: "",
-        confirm: "",
-      },
-    },
-  });
-  const validateUsername = useValidateUsername();
-  const validateEmail = useValidateEmail();
-  const validatePhoneNumber = useValidatePhoneNumber();
-  const validatePassword = useValidatePassword();
-  const required = useRequired();
-  const requirePassword =
-    !!form.watch("password.new") ||
-    !!form.watch("password.current") ||
-    !!form.watch("password.confirm");
-
-  const onSuccess = useCallback(() => {
-    alert("refresh");
-  }, []);
-
-  const onError = useCallback(
-    (error: Error) => {
-      toast.error({
-        title: intl("profile.update.error"),
-        description: error.message,
-      });
-    },
-    [intl, toast]
-  );
+  const { user, loading, fetching, error, refetch } = useUser();
 
-  const mutation = useUpdateUser({ onSuccess, onError });
-
-  const onSubmit = useCallback(
-    (data: IForm) => {
-      if (photo) console.log("upload photo");
-      if (!user) return;
-
-      mutation.mutate({
-        id: user.id,
-        payload: {
-          name: data.name && data.name !== user.name ? data.name : undefined,
-          email:
-            data.email && data.email !== user.email ? data.email : undefined,
-          phoneNumber: data.phoneNumber ? data.phoneNumber : undefined,
-          password:
-            data.password.new && data.password.current
-              ? {
-                  new: data.password.new,
-                  current: data.password.current,
-                }
-              : undefined,
-        },
-      });
-    },
-    [mutation, photo, user]
-  );
+  if (loading) {
+    return (
+      <div className="w-full h-full flex justify-center items-center">
+        <Loader variant="large" text={intl("settings.loading")} />;{" "}
+      </div>
+    );
+  }
 
-  const notifyComingSoon = useCallback(() => {
-    toast.success({
-      title: intl("settings.notifications.coming-soon.title"),
-      description: intl("settings.notifications.coming-soon.description"),
-    });
-  }, [intl, toast]);
+  if (error) {
+    return (
+      <div className="w-full h-full flex justify-center items-center">
+        <LoadingError retry={refetch.user} error={intl("settings.loading")} />
+      </div>
+    );
+  }
```

Comment:
This is not how we render the loading and the error states. It should be rendered inside the card. Please review the design.